### PR TITLE
The Great CI Cleanup 

### PR DESCRIPTION
**Pull request by BatiGencho**

### CODE REVIEW

**Comment by arminsabouri** [comment]
@birozuru when this is ready for review can you squash these commits?

**Comment by arminsabouri** [comment]
Do we need both push and pull triggers? I think the latter is probably enough

File: .github/workflows/dependencies.yml [hunk 1]
```diff
@@ -1,61 +0,0 @@
-# Runs `cargo update` periodically.
```
Comment (arminsabouri) [inline .github/workflows/dependencies.yml:1]: Should we delete this? Seems useful

File: .github/workflows/integration.yml [hunk 2]
```diff
@@ -1,86 +1,149 @@
...
-
-name: integration
+name: Botanix Integration Tests
 
 on:
   pull_request:
   merge_group:
   push:
-    branches: [main]
+    branches: [main, feature/mark/ci/quick-fix]
```
Comment (arminsabouri) [inline .github/workflows/integration.yml]: Should be main and feature/poa-consensus?

File: .github/workflows/lint.yml [hunk 3]
```diff
@@ -4,115 +4,115 @@ on:
   pull_request:
   merge_group:
   push:
-    branches: [main]
+    branches: [main, feature/mark/ci/quick-fix]
```
Comment (arminsabouri) [inline .github/workflows/lint.yml]: Same here: Should be main and feature/poa-consensus?

File: .github/workflows/unit.yml [hunk 4]
```diff
@@ -1,12 +1,10 @@
...
-
-name: unit
+name: Botanix Unit Tests
 
 on:
   pull_request:
   merge_group:
   push:
-    branches: [main]
+    branches: [main, feature/mark/ci/quick-fix]
```
Comment (arminsabouri) [inline .github/workflows/unit.yml]: Same here: Should be main and feature/poa-consensus?

File: .github/workflows/unit.yml [hunk 5]
```diff
@@ -19,20 +17,41 @@ concurrency:
...
         with:
           cache-on-failure: true
+          cache-all-crates: true
+
       - uses: taiki-e/install-action@nextest
+
+      - name: Start BTC Server
+        run: |
+          cd ./bin/btc-server && \
+          cargo run --bin btc-server -- \
```
Comment (arminsabouri) [inline .github/workflows/unit.yml]: Would it make sense to build the btc server once and then run `./target/release/binary`

File: .github/workflows/unit.yml [hunk 6]
```diff
@@ -63,31 +81,55 @@ jobs:
...
     if: always()
     needs: [test, state, doc]
-    timeout-minutes: 30
+    timeout-minutes: 90
     steps:
       - name: Decide whether the needed jobs succeeded or failed
         uses: re-actors/alls-green@release/v1
         with:
-          jobs: ${{ toJSON(needs) }}
+          jobs: ${{ toJSON(needs) }}
```
Comment (arminsabouri) [inline .github/workflows/unit.yml]: Missing eof

File: Cargo.toml [hunk 7]
```diff
@@ -200,7 +200,8 @@ ethers-core = { version = "2.0", default-features = false }
 ethers-providers = { version = "2.0", default-features = false }
 ethers-signers = { version = "2.0", default-features = false }
 ethers-middleware = { version = "2.0", default-features = false }
-discv5 = { git = "https://github.com/sigp/discv5", rev = "f289bbd4c57d499bb1bdb393af3c249600a1c662" }
+#discv5 = { git = "https://github.com/sigp/discv5", rev = "f289bbd4c57d499bb1bdb393af3c249600a1c662" }
```
Comment (arminsabouri) [inline Cargo.toml:203]: Could probably remove comment?

File: bin/btc-server/src/dkg.rs [hunk 8]
```diff
@@ -118,7 +118,7 @@ impl App {
         if frost_id == self.identifier {
             return Err(DKGError::CannotAddOwnDkgPackage);
         }
-        if self.frost_round1_dkg.as_ref().take().expect("valid dkg round1").1 == dkg_round1 {
```
Comment (arminsabouri) [inline bin/btc-server/src/dkg.rs:121]: Why did we remove take()? I'm pretty sure we want to explicity leave None here

File: bin/test-suite/src/macros.rs [hunk 9]
```diff
@@ -41,7 +41,7 @@ macro_rules! run_test {
             }
         }
 
-        $self.destroy_context().await;
+        //$self.destroy_context().await;
```
Comment (arminsabouri) [inline bin/test-suite/src/macros.rs]: Should we leave this commented out?

File: bin/test-suite/src/suite/consensus/mod.rs [hunk 10]
```diff
@@ -10,17 +10,94 @@ use port_killer::kill;
...
+            }
+        }
+        Err(err) => {
+            error!(
+                "Error attempting to kill btc-server process on port {:?} -> {:?}",
+                btc_server_port, err
+            );
+        }
+    }
+    /*
```
Comment (arminsabouri) [inline bin/test-suite/src/suite/consensus/mod.rs]: Could probably remove this?

File: crates/net/network/src/frost/protocol.rs [hunk 11]
```diff
@@ -163,234 +163,229 @@ impl Stream for FrostProtoConnection {
...
-                    // answer once the pong is received
-                    FrostPeerCommand::PingMessage { msg: _, response } => {
-                        //info!(">>>>>>>>> SENDING PING WITH MY AUTH INDEX {:?}", msg);
-                        this.pending_pong = Some(response);
-                        Poll::Ready(Some(
-                            FrostProtoMessage::ping_message(
-                                this.my_peer_id,
-                                this.my_authority_index,
-                            )
+        //loop {
```
Comment (arminsabouri) [inline crates/net/network/src/frost/protocol.rs]: Could delete this comment

File: .github/workflows/unit.yml [hunk 12]
```diff
@@ -22,13 +22,13 @@ concurrency:
...
   test:
     name: test / ethereum
-    runs-on: botanix-actions-runner-vm
+    runs-on: botanix-actions-runner-vm-1
     env:
       RUST_BACKTRACE: 1
     timeout-minutes: 90
     steps:
       - uses: actions/checkout@v4
-      - uses: taiki-e/install-action@nextest
```
Comment (arminsabouri) [inline .github/workflows/unit.yml]: is this pre-installed on the runner? other wise we need it for L42 
@birozuru

Comment (birozuru) [reply .github/workflows/unit.yml]: It is pre-installed already on the runner. The pipeline is in process

File: .github/workflows/deploy-btc-server-v1.yml [hunk 13]
```diff
@@ -0,0 +1,85 @@
+name: Build, Tag, and Deploy Docker BTC Server v1 Image to GCP
+
+on:
+  pull_request:
+    branches:
+      - feature/poA-consensus
+    types: [closed]
+  push:
```
Comment (arminsabouri) [inline .github/workflows/deploy-btc-server-v1.yml:8]: TODO to remove this later

File: .github/workflows/deploy-btc-server-v1.yml [hunk 14]
```diff
@@ -0,0 +1,85 @@
...
+      - name: Checkout code
+        uses: actions/checkout@v2
+
+      - name: Set up Docker
+        uses: docker/setup-buildx-action@v1
+
+      # NOTE: this operation should be replaced by a cloud or someother form of HSM once software is mainnet ready
+      # key.template.hex does not exist in the repo, it is a placeholder for the actual key
+    #   - name: Copy key hex
+    #     run: cp bin/btc-server/key.template.hex bin/btc-server/key.hex
```
Comment (arminsabouri) [inline .github/workflows/deploy-btc-server-v1.yml:26]: We can remove this comment as we no longer use key.hex
Fine to do in a another pr
